src/poetry/repositories/http_repository.py

```diff
@@ -401,7 +401,7 @@ def calculate_sha256(self, link: Link) -> str | None:
         return None
 
     def _get_response(self, endpoint: str) -> requests.Response | None:
-        url = self._url + endpoint
+        url = self.authenticated_url + endpoint
```

Comment:
I wonder if this is always correct? There is also https://github.com/python-poetry/poetry/blob/fee7d0d2a0d3170cac456f6505471278ccd09ee6/src/poetry/utils/authenticator.py#L196-L198

which seems to handle HTTP basic auth by adding a header if I understand correctly.